browser/pFa_test.go

```diff
@@ -0,0 +1,233 @@
+package browser
+
+import (
+	"github.com/vertgenlab/gonomics/dna/pDna"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fasta/pFasta"
+	"github.com/vertgenlab/gonomics/fileio"
+	"testing"
+)
+
+var PrintOneSetLinesTests = []struct {
+	LineLength    int
+	SetOfLinesIdx int
+	NumIters      int
+	LineA         []float32
+	LineC         []float32
+	LineG         []float32
+	LineT         []float32
+	Start         int
+	Records       []pFasta.PFasta
+	Out           string
+	SigFigs       int
+	DecimalPlaces int
+	LongestName   int
+	Expected      string
+}{{LineLength: 10,
+	SetOfLinesIdx: 0,
+	NumIters:      6,
+	LineA:         make([]float32, 30),
+	LineC:         make([]float32, 30),
+	LineG:         make([]float32, 30),
+	LineT:         make([]float32, 30),
+	Start:         2,
+	Records: []pFasta.PFasta{pFasta.PFasta{Name: "chr1",
+		Seq: []pDna.Float32Base{
+			pDna.Float32Base{
+				A: 0.23857,
```

Comment:
It seems like the pDna bases do not add up to 1, so they are likelihoods not probabilities. Maybe we should decide if we want to enforce probabilities, so each pDna base is "either {0 0 0 0} or all bases add up to 1"?

Reply:
We should have this discussion together, but we were thinking of not restricting the sum to 1, as we could theoretically store things other than valid probability vectors in this format. We could consider, for example, representing DNA sequences with uncertainty where the total magnitude of a vector is representative of another value, like conservation or chromatin accessibility. While we anticipate mostly using these for probabilities, I wanted to keep that option open